### ISSUE TITLE
MapView.Circle don't update when coordinate changes

### ISSUE THREAD

**Issue by yonahforst** [react-community/react-native-maps#283]
related to #187

I'm rendering my map like so: 

``` js
                <MapView
                ref='mapView'
                style={{flex: 1}}
                onPress={this._onPress.bind(this)}
                followsUserLocation={true}
                showsCompass={true}
                showsUserLocation={true} >
                    {this.props.onlineUsers.map(u => Marker(u, this.props))}
                    {this.props.selectedUser ? Circle(this.props.selectedUser) : null}
                </MapView>
```

and corresponding circle class:

``` js
const Circle = user => {
    console.log(user.location.coords.longitude)
    return (
    <MapView.Circle
        center={user.location.coords}
        radius={user.location.coords.accuracy}
        strokeColor='transparent'
        fillColor={wRedLighter}/>
    )
}
```

the logs show that the coordinates are changing. but the circle remains in the same place until i remove it and re-add it.

**Comment by MarcusSjolin** [comment]
I get the same problem when I try to have a radius in state, and if I have + / - buttons to increase / decrease the radius value I can see it change, however the circle won't change on the map. Seems to not rerender itself when there's a change in props.

**Comment by shanedevane** [comment]
I have this issue on IOS but not android.

**Comment by ecjs** [comment]
I actually saw this issue myself a little while ago as well.  Only tested this on iOS though.

**Comment by pixelgamma** [comment]
Same problem here on iOS simulator.

**Comment by yonahforst** [comment]
The underlying problem is that iOS doesn't re-render an MKOverlay just because its properties change. A simple solution is to cause react to remove and re-add the element when size or position change (you won't able to animate the change, and there will be a slight flicker)

How does react decide if an element should be removed? I tried setting a key that was based on the circles position, hoping that react will think that this is a new element. but that didn't work.

**Comment by yonahforst** [comment]
Ok, think i've got it. Pretty simple actually, just set the key to be a combination of the Circle's coordinate and radius. 

I'm doing something like this:
```js
const Circle = user => {
    return (
    <MapView.Circle
        key={(user.location.coords.longitude + user.location.coords.latitude + user.location.coords.accuracy).toString()}
        center={user.location.coords}
        radius={user.location.coords.accuracy}
        strokeColor='transparent'
        fillColor={wRedLighter}/>
    )
}
```
but you could also use some md5 or other hashing library to quickly calculate a hash for the entire object.

This causes the key to change every time the coordinates/radius changes, causing react to think it's a new element so the old one gets removed and the new one gets added

**Comment by pixelgamma** [comment]
Well done!

**Comment by ecjs** [comment]
Nailed it!

**Comment by felipegarcia92** [comment]
Thanks, it worked! 

However, isn't it a work-around? Shouldn't this issue be opened still?

**Comment by kanekotic** [comment]
i agree... this seems a workaround. and the change of center should be enough to update.

**Comment by yonahforst** [comment]
@kanekotic - i'll reopen the issue, incase they want to put in a more permanent fix

**Comment by kanekotic** [comment]
@yonahforst ok i am forking the repo to see if i can provide any fix for it :)

**Comment by NormalAndrew** [comment]
Clever workaround!  Any permanent fix to this yet?

**Comment by neilakoh** [comment]
the key thing works like a charm :)

**Comment by doniyor2109** [comment]
really helped

**Comment by macelii** [comment]
After twenty minutes debugging, I am glad I found this. :)
Thanks!

**Comment by lfkwtz** [comment]
This looks like it was fixed in #2101

**Comment by rborn** [comment]
Closing as per https://github.com/react-community/react-native-maps/issues/283#issuecomment-419176684
